From 43a0757fb33769d9ad4ca09e8f2ac572f9f6a491 Mon Sep 17 00:00:00 2001
From: mokaddem <sami.mokaddem@circl.lu>
Date: Fri, 6 Mar 2020 15:44:58 +0100
Subject: [PATCH] fix: [security] Fix reflected xss via unsanitized URL
 parameters

- As reported by an external pentest company on behalf of the Centre for Cyber security Belgium (CCB)
---
 app/View/Users/statistics_orgs.ctp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/app/View/Users/statistics_orgs.ctp b/app/View/Users/statistics_orgs.ctp
index 3bbf4c836f..6f22950ea2 100644
--- a/app/View/Users/statistics_orgs.ctp
+++ b/app/View/Users/statistics_orgs.ctp
@@ -12,7 +12,9 @@
                 'external' => array('selected' => false, 'text' => __('Known remote organisations')),
                 'all' => array('selected' => false, 'text' => __('All organisations'))
         );
-        $types[$scope]['selected'] = true;
+        if (isset($types[$scope])) {
+            $types[$scope]['selected'] = true;
+        }
     ?>
     <h4><?php echo __('Organisation list');?></h4>
     <p><?php echo __('Quick overview over the organisations residing on or known by this instance.');?></p>
